internal/services/report_processor.go

```diff
@@ -117,44 +117,56 @@ func ProcessReport(msg *kafka.Message) {
 				continue
 			}
 
-			usage_data_byte, err := kruize.Update_results(experiment_name, k8s_object)
-			if err != nil {
-				log.Error(err)
-				continue
-			}
-
-			for _, data := range usage_data_byte {
-
-				interval_start_time, _ := utils.ConvertStringToTime(data.Interval_start_time)
-				interval_end_time, _ := utils.ConvertStringToTime(data.Interval_end_time)
+			k8s_object_chunks := utils.SliceK8sObjectToChunks(k8s_object)
 
-				if workload_metrics, err := model.GetWorkloadMetricsForTimestamp(experiment_name, interval_end_time); err != nil {
-					log.Errorf("Error while checking for workload_metrics record: %s", err)
-					continue
-				} else if !reflect.ValueOf(workload_metrics).IsZero() {
-					log.Debugf("workload_metrics table already has data for interval_end time: %v.", interval_end_time)
+			for _, chunk := range k8s_object_chunks {
+				usage_data_byte, err := kruize.Update_results(experiment_name, chunk)
+				if err != nil {
+					log.Error(err, experiment_name)
 					continue
 				}
 
-				for _, container := range data.Kubernetes_objects[0].Containers {
-					container_usage_metrics, err := json.Marshal(container.Metrics)
+				for _, data := range usage_data_byte {
+
+					interval_start_time, err := utils.ConvertISO8601StringToTime(data.Interval_start_time)
 					if err != nil {
-						log.Errorf("Unable to marshal container usage data: %v", err)
+						log.Errorf("Unable to convert start time ISO8601 string to time: %s", err)
```

Comment:
```suggestion
						log.Errorf("Error for start time: %s", err)
```
As `err` object already includes the error message, no need of full message here.  To identify which field, you can include field name here.